test-restart.sh

```diff
@@ -23,7 +23,7 @@ wait_for_mysql
 echo "DB came back online; checking for clean shutdown"
 date
 docker logs "$MYSQL_CONTAINER" 2>&1 | grep "Shutdown complete"
-docker logs "$MYSQL_CONTAINER" 2>&1 | grep -qv "Crash recovery finished"
+docker logs "$MYSQL_CONTAINER" 2>&1 | grep -qvi "crash recovery finished"
```

Comment:
I think you said you already made this change so maybe it just hasn't been pushed.
```suggestion
(! docker logs "$MYSQL_CONTAINER" 2>&1 | grep -qi "crash recovery finished")
```
(optionally remove `-q` so that we see that the shutdown was unclean)